spring-integration-core/src/main/java/org/springframework/integration/aggregator/AbstractCorrelatingMessageHandler.java

```diff
@@ -673,7 +683,10 @@ protected void forceComplete(MessageGroup group) {
 
 	void remove(MessageGroup group) {
 		Object correlationKey = group.getGroupId();
-		this.messageStore.removeMessageGroup(correlationKey);
+		if (this.groupIds.contains(correlationKey)) {
```

Comment:
That's not logic for this method.
If we are here, we have to be already sure that it is our own group.
The real problem is definitely in the `ForceReleaseMessageGroupProcessor` implementation.
We have to check for its presence there.
All other logic should not be affected.

Please, make this `remove()` method `protected`.